### PR TITLE
FWUV-182 improve logging and add Tests to Post-Release Collection

### PR DESCRIPTION
**Pull request by jens-hammel**
Postman Collection is already imported into shared Postman workspace. See Confluence for Details.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
   </modules>
 
     <properties>
-        <revision>3.8.1</revision>
+        <revision>3.8.2</revision>
```
Comment (ingokuba) [inline pom.xml:28]: should also change CHANGELOG.md to explain bugfix change :thinking:

File: whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticator.java [hunk 2]
```diff
@@ -184,13 +185,16 @@ private boolean isAllowedIdP(AuthenticationFlowContext context, String clientId,
...
                 List<String> allowedIdPs = client.getListOfAllowedIdPs(clientId, apiRealm, apiClientId, apiClientSecret);
+                logger.tracef("Retrieve allowed IDPs using client %s, realm %s, secret %s from %s", apiClientId, apiRealm, apiClientSecret.substring(0, 3),
+                              client.getUrl());
+                logger.debugf("Retrieved allowed IDPs %s for sp %s from %s", allowedIdPs, providerId);
                 if (allowedIdPs != null && allowedIdPs.contains(providerId)) {
                     isAllowed = true;
                 }
-            } catch (IOException | URISyntaxException e) {
-                logger.errorf(e, "List of assigned IdPs could not fetched clientId=%s, providerId=%s, url=%s", clientId, providerId, client.getUrl());
+            } catch (IOException | URISyntaxException | JSONException e) {
```
Comment (ingokuba) [inline whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticator.java:196]: why catch this RuntimeException?

**Review by ingokuba — CHANGES_REQUESTED** [review]

File: whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticator.java (hunk 2 above)
Comment (jens-hammel) [reply whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticator.java:196]: to log it. We had auth issues which did not show up in the logs.

File: whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticator.java [hunk 3]
```diff
@@ -184,13 +185,16 @@ private boolean isAllowedIdP(AuthenticationFlowContext context, String clientId,
             try {
                 String apiRealm = getConfigEntry(context, WhitelistAuthenticatorFactory.AUTH_WHITELIST_REALM, context.getRealm().getName());
                 String apiClientId = getConfigEntry(context, WhitelistAuthenticatorFactory.AUTH_WHITELIST_CLIENT_ID, null);
-                String apiClientSecret = getConfigEntry(context, WhitelistAuthenticatorFactory.AUTH_WHITELIST_CLIENT_SECRET, null);
+                String apiClientSecret = getConfigEntry(context, WhitelistAuthenticatorFactory.AUTH_WHITELIST_CLIENT_SECRET, "n/a");
```
[the remark this answers is not in the document]
Comment (jens-hammel) [reply whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticator.java:188]: Changed title. Created new Ticket. Will create new MR for Postman collection.
A lot of effort for little change of logging...

**Pull request merged**